kyaml/fn/runtime/container/container.go

```diff
@@ -195,11 +195,23 @@ func (c *Filter) getCommand() (string, []string) {
 	// export the local environment vars to the container
 	for _, pair := range os.Environ() {
 		items := strings.Split(pair, "=")
-		if items[0] == "" || items[1] == "" {
+		if items[0] == "" || items[1] == "" || shouldEnvIgnored(items[0]) {
 			continue
 		}
 		args = append(args, "-e", items[0])
 	}
 	a := append(args, c.Image)
 	return "docker", a
 }
+
+// shouldEnvIgnored returns true if the environment variable key should be ignored
+// by the container runtime.
+func shouldEnvIgnored(envKey string) bool {
+	ignoreEnvKey := []string{"TMPDIR"}
+	for _, k := range ignoreEnvKey {
```

Comment:
In java I would use a set ignoreEnvKeys so I could quickly check ignoreEnvKeys.contains(envKey). Is there a way to do this in golang instead of iterating over the whole list.
Not a blocker for the PR since there's just one element to ignore anyway.

Reply:
Go has map, but in this case I think using map is a little bit too heavy because we have only one item.

Reply:
That's fair.